Document Nextflow syntax parser v2 features (#1444)
seqeralabs/docs: platform-enterprise_docs/launch/advanced.md

Replace Nextflow process commands with command [stubs](https://docs.seqera.io/nextflow/process#stub), where defined, before execution.

## Enable Nextflow syntax parser v2

> Use the new Nextflow language parser. Requires Nextflow 25.02.0-edge or later. Older runtimes ignore this setting.

Reviewer: ```suggestion
Use the new v2 Nextflow language parser. Requires Nextflow 25.02.0-edge or later. Older runtimes ignore this setting.
```

The v2 parser implements Nextflow's [strict syntax](https://nextflow.io/docs/latest/strict-syntax.html). Platform exports `NXF_SYNTAX_PARSER` to the launch environment:

- **Off (default)**: Workflows run with the v1 parser. Platform exports `NXF_SYNTAX_PARSER=v1`.
- **On**: Workflows run with the v2 parser. Platform exports `NXF_SYNTAX_PARSER=v2`.

The toggle only selects the parser. It does not change the Nextflow runtime version, the pipeline source, or any pipeline parameters.

> v2 becomes the default parser in Nextflow 26.04:

Reviewer: ```suggestion
The v2 parser becomes the default in Nextflow 26.04:
```

- **Before Nextflow 26.04**: Turn the toggle on to opt in to v2.

> - **From Nextflow 26.04**: v2 is the runtime default. Switch new pipelines on to match it, or turn the toggle off to pin a pipeline to v1.

Reviewer: ```suggestion
- **From Nextflow 26.04**: v2 is the runtime default. Turn the toggle off to pin a pipeline to v1.
```

Replace Nextflow process commands with command [stubs](https://docs.seqera.io/nextflow/process#stub), where defined, before execution.

## Enable Nextflow syntax parser v2

:::info

> The v2 parser is scheduled to become the **default** in Nextflow 26.04. Once that release is in use, this toggle has no effect and existing pipelines run with v2 regardless of the toggle's state.

Reviewer: ```suggestion
The v2 parser becomes the **default** in Nextflow 26.04. After this release, the toggle has no effect, and existing pipelines run with v2.
```

:::

> The v2 parser implements Nextflow's [strict syntax](https://nextflow.io/docs/latest/strict-syntax.html). Use the **Enable Nextflow syntax parser v2** toggle to select which Nextflow language parser is used at runtime. Existing pipelines continue to launch with v1. This toggle only selects the Nextflow language parser. It does not change the Nextflow runtime version, the pipeline source, or any pipeline parameters.

Reviewer: ```suggestion
The v2 parser implements Nextflow's [strict syntax](https://nextflow.io/docs/latest/strict-syntax.html). Use the **Enable Nextflow syntax parser v2** toggle to select which Nextflow language parser runs at launch. Existing pipelines continue to use v1 unless you turn the toggle on. This toggle does not change the Nextflow runtime version, the pipeline source, or any pipeline parameters.
```

Reviewer: I'm surprised the toggle has no effect on >=26.04. In Nextflow, it would revert to v1 when you set the env variable, as suggested by the **On** point below.

Reviewer: This may be a clarity thing. Versions prior to 26.04 use v1 by default unless you toggle it on. 26.04 onwards, v2 is used by default unless the toggle is off?

Reviewer: But this does counter the "regardless of the toggle's state" noted above

> - **Off (default)**: Workflows run with the v1 parser, which accepts the full Groovy syntax that Nextflow has historically supported. Platform exports `NXF_SYNTAX_PARSER=v1` to the Nextflow execution environment.

Reviewer: ```suggestion
- **Off (default)**: Workflows run with the v1 parser, which accepts the full Groovy syntax that Nextflow has historically supported. Platform exports `NXF_SYNTAX_PARSER=v1`.
```

- **On**: Workflows run with the v2 parser. Platform exports `NXF_SYNTAX_PARSER=v2`.

:::caution

> Test pipelines on a non-production run before enabling the parser. The [Nextflow strict syntax documentation](https://nextflow.io/docs/latest/strict-syntax.html) describes which constructs are accepted.

Reviewer: ```suggestion
Test pipelines on a non-production run before enabling the parser. The [Nextflow strict syntax documentation](https://nextflow.io/docs/latest/strict-syntax.html) lists the accepted constructs.
```

:::

### How the value resolves

The launch form inherits its initial value from the pipeline. You can override it for a single run without changing the pipeline:

| Pipeline toggle | Launch form toggle (default) | Launched workflow | Pipeline stored value after launch |
|-----------------|------------------------------|-------------------|------------------------------------|
| Off             | Off                          | v1                | Unchanged (off)                    |
| On              | On (inherited)               | v2                | Unchanged (on)                     |
| On              | Off (overridden at launch)   | v1                | Unchanged (on)                     |
| Off             | On (overridden at launch)    | v2                | Unchanged (off)                    |

> Changing the toggle on the pipeline edit form creates a new pipeline version, the same way a change to **Pull latest** or **Stub run** does.

Reviewer: ```suggestion
Changing the toggle on the pipeline edit form creates a new pipeline version, like changes to **Pull latest** or **Stub run**.
```
